### ISSUE TITLE
Internal compiler error with Evaluate() in CODEBLOCK with /memvar+ in VFP dialect

### ISSUE THREAD

**Issue by cpyrgas** [X-Sharp/XSharpPublic#1043]
An ICE is caused by using Evaluate() in a codeblock in the VFP dialect:

```
// VFP dialect, /memvar+
CLASS TestClass
    METHOD TestMethod()
    	LOCAL cb := {|| Evaluate("1+2") } // ICE
    	SET FILTER TO Evaluate("1==1") // ICE
     END METHOD 
END CLASS


CompileGeneratedCode.prg(1,311): error XS1558: 'fox.Exe.Functions' does not have a suitable static 'Start' function
error XS9999: An internal compiler error has occurred: 'The given key was not present in the dictionary.',    at System.ThrowHelper.ThrowKeyNotFoundException()
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.FramePointer(SyntaxNode syntax, NamedTypeSymbol frameClass)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriterWithStackGuard.VisitExpressionWithoutStackGuard(BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeVisitor.VisitExpressionWithStackGuard(Int32& recursionDepth, BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriter.DoVisitList[T](ImmutableArray`1 list)
   at LanguageService.CodeAnalysis.XSharp.Symbols.MethodToClassRewriter.VisitCall(BoundCall node)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.VisitCall(BoundCall node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriterWithStackGuard.VisitExpressionWithoutStackGuard(BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeVisitor.VisitExpressionWithStackGuard(Int32& recursionDepth, BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriter.DoVisitList[T](ImmutableArray`1 list)
   at LanguageService.CodeAnalysis.XSharp.Symbols.MethodToClassRewriter.VisitCall(BoundCall node)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.VisitCall(BoundCall node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriterWithStackGuard.VisitExpressionWithoutStackGuard(BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeVisitor.VisitExpressionWithStackGuard(Int32& recursionDepth, BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.RewriteSequence(BoundSequence node, ArrayBuilder`1 prologue, ArrayBuilder`1 newLocals)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.VisitSequence(BoundSequence node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriterWithStackGuard.VisitExpressionWithoutStackGuard(BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeVisitor.VisitExpressionWithStackGuard(Int32& recursionDepth, BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriter.VisitPassByCopy(BoundPassByCopy node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriterWithStackGuard.VisitExpressionWithoutStackGuard(BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeVisitor.VisitExpressionWithStackGuard(Int32& recursionDepth, BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriter.VisitReturnStatement(BoundReturnStatement node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriterWithStackGuard.Visit(BoundNode node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriter.VisitSequencePoint(BoundSequencePoint node)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.RewriteBlock(BoundBlock node, ArrayBuilder`1 prologue, ArrayBuilder`1 newLocals)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.VisitBlock(BoundBlock node)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.RewriteLambdaOrLocalFunction(IBoundLambdaOrFunction node, ClosureKind& closureKind, NamedTypeSymbol& translatedLambdaContainer, SynthesizedClosureEnvironment& containerAsFrame, BoundNode& lambdaScope, DebugId& topLevelMethodId, DebugId& lambdaId)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.RewriteLambdaConversion(BoundLambda node)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.VisitConversion(BoundConversion conversion)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriterWithStackGuard.VisitExpressionWithoutStackGuard(BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeVisitor.VisitExpressionWithStackGuard(Int32& recursionDepth, BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriter.DoVisitList[T](ImmutableArray`1 list)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriter.VisitObjectCreationExpression(BoundObjectCreationExpression node)
   at LanguageService.CodeAnalysis.XSharp.Symbols.MethodToClassRewriter.VisitObjectCreationExpression(BoundObjectCreationExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriterWithStackGuard.VisitExpressionWithoutStackGuard(BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeVisitor.VisitExpressionWithStackGuard(Int32& recursionDepth, BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriter.VisitConversion(BoundConversion node)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.VisitConversion(BoundConversion conversion)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriterWithStackGuard.VisitExpressionWithoutStackGuard(BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeVisitor.VisitExpressionWithStackGuard(Int32& recursionDepth, BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriter.VisitConversion(BoundConversion node)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.VisitConversion(BoundConversion conversion)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriterWithStackGuard.VisitExpressionWithoutStackGuard(BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeVisitor.VisitExpressionWithStackGuard(Int32& recursionDepth, BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriter.DoVisitList[T](ImmutableArray`1 list)
   at LanguageService.CodeAnalysis.XSharp.Symbols.MethodToClassRewriter.VisitCall(BoundCall node)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.VisitCall(BoundCall node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriterWithStackGuard.VisitExpressionWithoutStackGuard(BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeVisitor.VisitExpressionWithStackGuard(Int32& recursionDepth, BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.Symbols.MethodToClassRewriter.VisitAssignmentOperator(BoundAssignmentOperator node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriterWithStackGuard.VisitExpressionWithoutStackGuard(BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeVisitor.VisitExpressionWithStackGuard(BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeVisitor.VisitExpressionWithStackGuard(Int32& recursionDepth, BoundExpression node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriter.VisitExpressionStatement(BoundExpressionStatement node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriterWithStackGuard.Visit(BoundNode node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriter.VisitSequencePointWithSpan(BoundSequencePointWithSpan node)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.RewriteBlock(BoundBlock node, ArrayBuilder`1 prologue, ArrayBuilder`1 newLocals)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.IntroduceFrame(BoundNode node, ClosureEnvironment env, Func`3 F)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.VisitBlock(BoundBlock node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriterWithStackGuard.Visit(BoundNode node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriter.VisitTryStatement(BoundTryStatement node)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.RewriteBlock(BoundBlock node, ArrayBuilder`1 prologue, ArrayBuilder`1 newLocals)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.VisitBlock(BoundBlock node)
   at LanguageService.CodeAnalysis.XSharp.BoundTreeRewriterWithStackGuard.Visit(BoundNode node)
   at LanguageService.CodeAnalysis.XSharp.ClosureConversion.Rewrite(BoundStatement loweredBody, NamedTypeSymbol thisType, ParameterSymbol thisParameter, MethodSymbol method, Int32 methodOrdinal, MethodSymbol substitutedSourceMethod, ArrayBuilder`1 lambdaDebugInfoBuilder, ArrayBuilder`1 closureDebugInfoBuilder, VariableSlotAllocator slotAllocatorOpt, TypeCompilationState compilationState, DiagnosticBag diagnostics, HashSet`1 assignLocals)
   at LanguageService.CodeAnalysis.XSharp.MethodCompiler.LowerBodyOrInitializer(MethodSymbol method, Int32 methodOrdinal, BoundStatement body, SynthesizedSubmissionFields previousSubmissionFields, TypeCompilationState compilationState, Boolean instrumentForDynamicAnalysis, DebugDocumentProvider debugDocumentProvider, ImmutableArray`1& dynamicAnalysisSpans, DiagnosticBag diagnostics, VariableSlotAllocator& lazyVariableSlotAllocator, ArrayBuilder`1 lambdaDebugInfoBuilder, ArrayBuilder`1 closureDebugInfoBuilder, StateMachineTypeSymbol& stateMachineTypeOpt)
   at LanguageService.CodeAnalysis.XSharp.MethodCompiler.CompileMethod(MethodSymbol methodSymbol, Int32 methodOrdinal, ProcessedFieldInitializers& processedInitializers, SynthesizedSubmissionFields previousSubmissionFields, TypeCompilationState compilationState)



```

**Comment by RobertvanderHulst** [comment]
Fixes #1049 X-Sharp/XSharpPublic/issues/1049

**Comment by cpyrgas** [comment]
Confirmed fixed.
